### ISSUE TITLE
🐛 [Bug Report] New Characters start naked

### ISSUE THREAD

**Issue by milestorme** [cmangos/issues#2317]
## 🐛 Bugreport
new characters start naked after first login ,  i installed al updates running InstallFullDB.sh with correct path to core folder no errors when running. fresh world fresh core absolutely no modifications
![WoWScrnShot_111720_173640](https://user-images.githubusercontent.com/44911744/99354951-82b90d00-28fb-11eb-9acd-c73fd3088147.jpg)

### Expected behavior
characters should start with cloths and hearthstone

### Version & Environment

Client Version: 
"2.4.3" (TBC)

CMaNGOS Repo & Commit Hash: 
  - [TBC](https://github.com/cmangos/mangos-tbc/tree/279fe2fe09a522338b15098f91dbaa285eb90cf5)

Database Repo & Commit Hash: 
https://github.com/cmangos/tbc-db/tree/ae833c5eb22f17f610a18cd29e910f6524c7487c

Operating System: 
Win 10
### Steps to reproduce
1. Create character 
2. Log in

### Crashlog

- None

**Comment by MantisLord** [comment]
Keeping this open for now as we want to keep an eye on this due to the item_instance transfer.
@milestorme mentioned his problem was fixed in discord, but there may be others experiencing similar things and just want to be sure we're okay here.

**Comment by canidae** [comment]
I recently upgraded both core & database (classic), as well as build tools and MySQL. All characters are clothed upon character selection, but entering the world with any of the characters cause them to lose all their clothes.
Creating new characters does not result in them entering the world without clothes.

There were some database errors when upgrading, but I didn't take much note as I also upgraded MySQL and assumed incompatibility.
I may attempt to upgrade another old database later this week, I'll see if that also results in this error.

**Comment by milestorme** [comment]
The reason I closed it was because after extraction of fresh DBC files from the latest build my issue resolved itself and characters had items and cloths on entry to world

**Comment by Daedalus-code** [comment]
Stuck at updating from sql/updates/characters/z2765_01_characters_item_instance_data_drop.sql
![stuck](https://user-images.githubusercontent.com/59256533/100095308-8fd57f00-2e5a-11eb-9a31-b4572a3416e0.png)
🤔 

It changes "**required_z2764_01_characters_ahbot_items**" into "**required_z2765_01_characters_item_instance_data_drop**" in "**character_battleground_data**" table, leaves a "**uint32toint32**" that did not get dropped, the update did not go all the way.

**Comment by canidae** [comment]
For my part it seems like this happened because of an incomplete upgrade of MySQL.
When attempting to apply a patch to the `classiccharacters` database I got this error:
```
root@15c84a434ac0:~# mysql classiccharacters < z2765_01_characters_item_instance_data_drop.sql
ERROR 1805 (HY000) at line 24: Column count of mysql.proc is wrong. Expected 21, found 20. The table is probably corrupted
```
A quick google gave me this answer: https://stackoverflow.com/questions/16177465/column-count-of-mysql-proc-is-wrong-expected-20-found-16-the-table-is-probabl

**TL;DR:**
Think error is due to upgrading MySQL?
* Dump data to SQL file before upgrading: `mysqldump -u root -p --all-databases > alldb.sql`
* Upgrade MySQL
* Import databases: `mysql -u root -p < alldb.sql`
* Fix tables (without `--force` it says everything is fine, but it wasn't for me, the `--force` fixed it):  `mysql_upgrade --force`

**Comment by Daedalus-code** [comment]
Yeah tried `mysql_upgrade --force` to no avail ): Same error as picture posted above 🤔
```
"Incorrect integer value: '' for column `classiccharacters_normal`.`item_instance`.`itemEntry` at row 1"
```

**Comment by diluh** [comment]
@MantisLord @henhouse
https://github.com/cmangos/mangos-tbc/commit/2cd35410968cc045bdefdfba7cd53c7b238f0080

Do you think this commit is causing any problems with the items of the characters? i have an open server, and i am having serious problems players lose several items, bags full of items and new naked characters.

when I restart the server it returns to normal but the characters lose their items and it does not appear in the post office, even restarting the server a while later it happens again.

**Comment by Booggmz** [comment]
I got solution for corrupted (after unsuccessful update) characters DB:

1) create  mass pdump command:
```c++
bool ChatHandler::HandlePDumpWriteAllCommand(char* args)
{
    QueryResult* result = CharacterDatabase.PQuery("SELECT guid FROM characters");
    if (result)
    {
        do {
            Field *fields = result->Fetch();
            uint32 lowguid = fields[0].GetUInt32();


            ObjectGuid guid = ObjectGuid(HIGHGUID_PLAYER, lowguid);
            std::string file = guid.GetString();

            if (!sObjectMgr.GetPlayerAccountIdByGUID(guid)) {
                PSendSysMessage(LANG_PLAYER_NOT_FOUND);
                SetSentErrorMessage(true);
                return false;
            }

            PlayerDumpWriter().WriteDump(file, lowguid);
            PSendSysMessage(LANG_COMMAND_EXPORT_SUCCESS, lowguid);
        }
        while (result->NextRow());
    }
}
```
and register it:
```
  static ChatCommand pdumpCommandTable[] =
    {
        { "load",           SEC_ADMINISTRATOR,  true,  &ChatHandler::HandlePDumpLoadCommand,           "", nullptr },
        { "write",          SEC_ADMINISTRATOR,  true,  &ChatHandler::HandlePDumpWriteCommand,          "", nullptr },
        { "writeall",       SEC_ADMINISTRATOR,  true,  &ChatHandler::HandlePDumpWriteAllCommand,          "", nullptr },
        { nullptr,          0,                  false, nullptr,                                        "", nullptr }
    };
```

then startup core with corrupted db and run `pdump writeall`

after pdump done, go to core path and batch all chardump files to clean characters db

I renamed all files to .sql and import over mysql < *.sql with error ignoring.
Imported over 2k characters and problem gone.

sry 4 my English, Russian version here https://cmangos.ru/d/8-massovyy-damp-personazhey xD

**Comment by canidae** [comment]
We discovered an issue with the migration SQL script. One of the fields ends up with `DEFAULT NULL` instead of `DEFAULT 0`, and that seems to prevent items from being stored.
This is only an issue for migrated databases, fresh databases would not see this error.

For those who have an existing database, you can correct the issue with the following command (assuming MySQL database) in database `classiccharacters`:
`ALTER TABLE item_instance ALTER duration SET DEFAULT 0;`

**Comment by diluh** [comment]
I tried to use this command nothing else happened it gives error Player Not Found can you help me?

**Comment by diluh** [comment]
even with the DEFAULT 0 change, the items are still disappearing today after 3 server crashes when restarting the server characters still losing items

**Comment by Daedalus-code** [comment]
Still stuck at this ```"Incorrect integer value: '' for column `classiccharacters_normal`.`item_instance`.`itemEntry` at row 1"```  🤣 🦖

**Comment by Booggmz** [comment]
> Still stuck at this `` "Incorrect integer value: '' for column `classiccharacters_normal`.`item_instance`.`itemEntry` at row 1" ``

its ok, item entry cant be empty :smile_cat:  check your data and converter scipt (itemEntry getting to single select)

**Comment by jimmybrancaccio** [comment]
@milestorme Are you still having this issue?

**Comment by milestorme** [comment]
No

**Comment by milestorme** [comment]
So I did get this issue today after the server crashed , I was able to fix it by rerunning the InstallFullDB.sh

**Comment by Valkirie** [comment]
Issue appears to be random. I'm having it from time to time after a server restart. Another restart could or could not fix it.
Definitely **not Closed**.

**Comment by milestorme** [comment]
@jimmybrancaccio can we re-open this please

**Comment by diluh** [comment]
I advise you to revert this change https://github.com/cmangos/mangos-tbc/commit/2cd35410968cc045bdefdfba7cd53c7b238f0080 serious problems with item_instance unsaved items, after any accident happens several items will be lost and are not recovered.  at some point if you type .saveall the server has an accident and again people lose item

**Comment by diluh** [comment]
I've tried all the solutions mentioned above and others you can imagine and nothing has resolved, so if you want to resolve reverse this change

**Comment by Daedalus-code** [comment]
> Yeah tried `mysql_upgrade --force` to no avail ): Same error as picture posted above 🤔
> 
> ```
> "Incorrect integer value: '' for column `classiccharacters_normal`.`item_instance`.`itemEntry` at row 1"
> ```

This had a row that had ```guid``` as 0, looked different from every other rows, deleted that row.
Tried updating with the 2765, got this error: ```'_INDEX(`data`,' ',45+1))-length(SUBSTRING_INDEX(data,' ',45))-1);```  then I went along with update 2770 and that went fine with no errors, restart server and login and everything seems fine 👀 been playing and levelling , got new gear, logout, login and it still looks good.

**Comment by jimmybrancaccio** [comment]
Looks like @MantisLord was quicker then I 😝 Hopefully we can figure out what is going on here and get it resolved.

**Comment by d3rrial** [comment]
When I set the mangosd.conf parameter `SkipCinematics` to `1` all characters created after that start naked.
When the value is set to `0`, new characters have their proper starting gear.

**Comment by insunaa** [comment]
Levi and I have been trying to debug this over the weekend and our strongest lead is GUID collisions during item creation. It doesn't seem to be an issue during regular playing, but once the new item with the colliding GUID is trying to get saved to the Database, the database silently rejects this, because (this is pure speculation!) it's using `REPLACE INTO`, but the table has 2 keys (1 unique `PRI` (the guid) and 1 `MUL` (the owner_guid)). Since owner_guid+guid together are unique, it's a valid database key, *but* because `guid` is a unique row by itself, it fails. So far for my hypothesis, but I actually couldn't find any error messages that support that from the Database.

What I did to test this was to change the `REPLACE INTO` statement to two separate statements, one `DELETE FROM x WHERE guid` and two an `INSERT INTO` statement. Now if guid collisions are a problem this will lead to random items from random characters to disappear, and on my own server so far this hasn't happened, but on someone elses server who tested it, the behavior of disappearing items changed (for example one player only *partially* lost new items, instead of completely)

What I could confirm without a doubt is that guids that are already in use are being re-issued, so I think there's an issue with the way guids are generated.

**Comment by canidae** [comment]
I'm inclined to support the theory that GUID's are reused. I've had a server running for a long time and there has been no issues, but suddenly new items started disappearing upon logout (i.e. items were not persisted in the database).
This is rather difficult to debug, as after adding some debug logging and restarting the server I was no longer able to reproduce the issue.
I'll keep the server running for a while with the following extra logging (in Item.cpp):
```
        case ITEM_NEW:
        case ITEM_CHANGED:
        {
            // DEBUG: Disappearing items
            if (uState == ITEM_NEW)
                sLog.outError("New item(%u) GUID: %u", GetEntry(), guid);
            // DEBUG: END
            static SqlStatementID insItem;
            SqlStatement stmt = CharacterDatabase.CreateStatement(insItem, uState == ITEM_NEW ?
                "REPLACE INTO item_instance (owner_guid, itemEntry, creatorGuid, giftCreatorGuid, count, duration, charges, flags, enchantments, randomPropertyId, durability, playedTime, text, guid) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?)" :
                "UPDATE item_instance SET owner_guid = ?, itemEntry = ?, creatorGuid = ?, giftCreatorGuid = ?, count = ?, duration = ?, charges = ?, flags = ?, enchantments = ?, randomPropertyId = ?, durability = ?, playedTime = ?, text = ? WHERE guid = ?"
            );
```
If items start disappearing again it's a simple matter of just creating an item and relogging to verify that the item is gone. Then I can check which GUID it received and whether that GUID exists in the database.

**Comment by insunaa** [comment]
Without the bug currently being active on my testserver, I still find it weird that when I create a new character that the GUIDs advance in steps of two, instead of steps of 1 as I would expect they should, given that Guid.Generate() just returns guid++
![image](https://user-images.githubusercontent.com/3718129/108199519-72444880-711d-11eb-89cb-ffda3d246463.png)

**Comment by mathias-bloodmoon** [comment]
I have this bug on my server now as well since today (28.02.2021). 

Two observations: 

[1]
Before the bug occured I had emptied the tables "item_instance" and "character_inventory" and my users were able to add new items without any problem for 1 day. Then today when the bug occured I saw in the Database that in table "item_instance" the last entry was 1701 but the IDs the game produced for new item instances were way higher - in the 1900s . I was expecting the game to raise the highest ID in the table by "1" which would result in ID = 1702

[2] 
I activated full mySQL logging and gave a character 2 new items. Upon logging out the game send 4 SQL statements regarding those items: 
2x INSERT into character_inventory with a new ID in the 1900s - this seems correct
2x UPDATE into item_instance with the same IDs. This UPDATE of course did nothing because there was no entry with that IDs since it was new items. - Why the update?  

Here are the SQL statements the game did upon logging out:
`1288 Execute	INSERT INTO character_inventory (guid,bag,slot,item,item_template) VALUES (1, 0, 23, 1908, 21376)`

`1288 Execute	UPDATE item_instance SET owner_guid = 1, itemEntry = 21376, creatorGuid = 0, giftCreatorGuid = 0, count = 1, duration = 0, charges = '0 0 0 0 0 ', flags = 1, enchantments = '0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ', randomPropertyId = 0, durability = 85, itemTextId = 0 WHERE guid = 1908`

 `1288 Execute	INSERT INTO character_inventory (guid,bag,slot,item,item_template) VALUES (1, 0, 24, 1909, 19373)`
		  
`1288 Execute	UPDATE item_instance SET owner_guid = 1, itemEntry = 19373, creatorGuid = 0, giftCreatorGuid = 0, count = 1, duration = 0, charges = '0 0 0 0 0 ', flags = 1, enchantments = '0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 ', randomPropertyId = 0, durability = 85, itemTextId = 0 WHERE guid = 1909`

Conclusion: 
For me the bug is that the game thinks that the item already exists , therefor UPDATe is necessary although the item is new. So the bug must be somewhere in the ITEM handling in the process which defines New items vs updated items
---
Disclaimer: I guess my assessements might be too non-technical or maybe even stupid but bare with me, I am not a C programmer and I do not know the game very well. Just trying to help.

**Comment by mathias-bloodmoon** [comment]
UPDATE: 
I added the patch from "Mein Name ist Hase" which changes function SaveToDB() in "item.cpp" so that in case of ITEM_NEW and ITEM_CHANGED only REPLACE is used in the SQL statement and no longer UPDATE. 
--> Result: Now new items for characters stay in inventory after logout and login.

**Comment by insunaa** [comment]
The patch in question is
```patch
diff --git a/src/game/Entities/Item.cpp b/src/game/Entities/Item.cpp
index a4e0748e5..cf8d9b980 100644
--- a/src/game/Entities/Item.cpp
+++ b/src/game/Entities/Item.cpp
@@ -302,9 +302,8 @@ void Item::SaveToDB()
         case ITEM_CHANGED:
         {
             static SqlStatementID insItem;
-            SqlStatement stmt = CharacterDatabase.CreateStatement(insItem, uState == ITEM_NEW ?
-                "REPLACE INTO item_instance (owner_guid, itemEntry, creatorGuid, giftCreatorGuid, count, duration, charges, flags, enchantments, randomPropertyId, durability, itemTextId, guid) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?)" :
-                "UPDATE item_instance SET owner_guid = ?, itemEntry = ?, creatorGuid = ?, giftCreatorGuid = ?, count = ?, duration = ?, charges = ?, flags = ?, enchantments = ?, randomPropertyId = ?, durability = ?, itemTextId = ? WHERE guid = ?"
+            SqlStatement stmt = CharacterDatabase.CreateStatement(insItem,
+                "REPLACE INTO item_instance (owner_guid, itemEntry, creatorGuid, giftCreatorGuid, count, duration, charges, flags, enchantments, randomPropertyId, durability, itemTextId, guid) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?)"
             );
 
             stmt.addUInt32(GetOwnerGuid().GetCounter());
```
and so far there is no evidence that it makes a difference

**Comment by canidae** [comment]
Whether it fixes the bug or not, it's a change that might as well be done. I'm guessing it once used to be `INSERT INTO` rather than `REPLACE INTO`, but with `REPLACE INTO` the `UPDATE` is superfluous.
This issue has still not reappeared on my server. It appeared for some few days, but restarting the server caused it to go away and never reappear.

**Comment by insunaa** [comment]
There are 2 reasons for using UPDATE instead of REPLACE INTO:
1) Update is less resource intensive
2) Replace triggers cascading deletes, Update does not. Example: Table 1 has a Secondary Key that points to Table 2's Primary key, if you do Replace in Table 2, the row in Table 1 that refers to table 2 will also be deleted *but it will not be recreated*

This Patch is ***for testing purposes only***
It doesn't fix this issue and probably creates other issues down the line. It can be used to evaluate Bloodmoon's reasoning, nothing more.

**Comment by canidae** [comment]
That's a valid point. I was thinking of `UPSERT` rather than `REPLACE`.

**Comment by sebeksd** [comment]
Hi, today morning this bug also showed up on my server. Server is using clean WorldDB (InstallFullDB+Core path) but character and realm DB are updated from 10 year old mangos DB.
Server was running for less then two days, yesterday there was like 2 accounts logged for 30min, rest of the time server was empty.
Today friend of my created character on new account (account was created yesterday using console) and his new character was naked after playing for few minutes he logged out and logged in, and his character still had looted items in bags (at least that what he said).
After server restart he lost all looted items but after restart creating new characters is fine, all characters have correct items on create and after restarting mangosd.
Before first restart I also created character on my old (10y) account and it was naked, after restart everything was fine.
There was no "serious" gaming when bug appeared, few mobs killed near starting locations.
No errors in logs (only typical ones).

Maybe there is some memory corruption that cause this erratic behavior ?

**Comment by insunaa** [comment]
Bloodmoon has just confirmed that the bug re-occurred even with the REPLACE/UPDATE patch in place, so that's probably not the cause of the issue either.
It has taken over a week to re-occur, tho, so can't be 100% sure about this, either.

**Comment by TheOzzOne** [comment]
I am also experiencing this bug; happy to help troubleshoot in any way that I can.

**Comment by canidae** [comment]
Try out the changes @insunaa recently did (https://github.com/cmangos/mangos-tbc/pull/441) and report back.

**Comment by MilkmanMHOC** [comment]
Totally new here, but also can report new characters on my new server were naked.  A server shutdown and restart fixed the issue... for now....

**Comment by insunaa** [comment]
@MilkmanMHOC The fix from https://github.com/cmangos/mangos-tbc/pull/441 should work with classic and wotlk too. The reason why the PR only exists on tbc is because I only work on the tbc core and not the others.

**Comment by MilkmanMHOC** [comment]
Ok will try that.  Thank you!

**Comment by canidae** [comment]
> @MilkmanMHOC The fix from [cmangos/mangos-tbc#441](https://github.com/cmangos/mangos-tbc/pull/441) should work with classic and wotlk too. The reason why the PR only exists on tbc is because I only work on the tbc core and not the others.

Just a heads up:
The WotLK `item_instance` is not identical to TBC `item_instance`, hence the patch from @insunaa cannot be directly copy-pasted to WotLK core. Here's a similar change for WotLK (line 306-311 in Entities/Item.cpp):
```
static SqlStatementID insItem, updItem;

SqlStatement stmt = CharacterDatabase.CreateStatement(uState == ITEM_NEW ? insItem : updItem, uState == ITEM_NEW ?
        "REPLACE INTO item_instance (owner_guid, itemEntry, creatorGuid, giftCreatorGuid, count, duration, charges, flags, enchantments, randomPropertyId, durability, playedTime, text, guid) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?)" :
        "UPDATE item_instance SET owner_guid = ?, itemEntry = ?, creatorGuid = ?, giftCreatorGuid = ?, count = ?, duration = ?, charges = ?, flags = ?, enchantments = ?, randomPropertyId = ?, durability = ?, playedTime = ?, text = ? WHERE guid = ?"
);
```

**Comment by insunaa** [comment]
The fix was added to the cores by cyberium.
Please keep your eyes open for the issue re-occurring, on the chance that this didn't address the root cause yet.

**Comment by cyberium** [comment]
Some change have been done it would be nice to continue to give us feedback on this.

I suspect that the item_instance is in bad state after https://github.com/cmangos/mangos-tbc/commit/2cd35410968cc045bdefdfba7cd53c7b238f0080
So its known that after that change you could have this issue one time for each characters.
But after that if you have it more than once on same character its not normal and you can report it here.

Be sure you run latest version before new report.

**Comment by diluh** [comment]
Hello all i come informs that the changes made by @cyberium still does not fix the problems with the items. 

I have an online server and after applying the changes to test stayed for 1 day and appeared some problems.

Items still disappear more they are now recovered through the post office, players have tried to send items to other players by the post office and fail.

Here's a video of a player who reported the Bug. The items then go to the post office can not be recovered and are multiplying at all times.

**Demonstration in video**

https://gofile.io/d/bS6M0l



![image](https://user-images.githubusercontent.com/46387591/111859045-a24d5a00-891c-11eb-9624-5e683a94df96.png)

**Comment by namreeb** [comment]
Does anyone know how to cause this to happen?

**Comment by insunaa** [comment]
@namreeb the original issue was caused by a race condition, you could force it to occur if the first thing you did after server start was enchant an item and log out, that would pre-bake the SQL query used to save items to database into an `UPDATE` query, instead of a `REPLACE INTO` query.

Sometimes this would happen naturally by way of this having been a race condition, so if it had worked properly before, because the query was baked to be `REPLACE INTO` it could switch around to becoming `UPDATE` again and then all new items couldn't be saved anymore, because there's no row it could update.

In short: For a quick way to force it to happen give your character a sharpening stone and a sword, log out, restart the server log in, use the sharpening stone, log out again and from then onward all new items would no longer be saved to database.

**Comment by namreeb** [comment]
Thanks @insunaa that is helpful.  Restating it to make sure I understood.
1. Start server
2. Login, create sharpening stone and sword in inventory
3. Logout
4. Restart server
5. Login, apply sharpening stone to sword
6. Logout
7. Bug is triggered

I did this and the bug did not occur.  Presumably this is because of the [merge request](https://github.com/cmangos/mangos-tbc/pull/441) that was made to fix the issue.  But I'm not clear on whether it fixed the issue or just made it less common (and broke your example on how to trigger it).

**Comment by insunaa** [comment]
That is exactly how to trigger the bug before the fix.
The code explanation for why it worked is that the `statementID` stores the actual prepared statement query. Before the fix this was a `static` variable, so it couldn't be rewritten by subsequent assignments. The `statementID` is then re-used by the other SqlStatement object. If it's pre-loaded with the `UPDATE` query, it can never save new objects.

**Comment by namreeb** [comment]
I'm sorry I hadn't noticed or read the conversation in the MR.  Nor did I realize that the MR came from you.  Thanks for that, by the way!  So this might be a stupid question, but is the bug fixed now?  I had it happen to me a few days ago, but I'm also operating on a branch and I am not positive whether it happened before or after I did a recent rebase off master.

**Comment by diluh** [comment]
I applied the fix on my server and happened Bug related to the Post Office, the items disappear goes to the post office and get buggy there, and can pick up the items infinitely because they multiply and some can not pick up

**Comment by insunaa** [comment]
Item disappearing and going to the post office can't be related to this bug, because with this bug items are treated as if they never existed, so they can't be restored by post-office stuff either

**Comment by diluh** [comment]
@insunaa I missed a full day of server had to return backup to be able to solve the problem.

After I apply the change https://github.com/cmangos/mangos-tbc/commit/5c721e3a380cb03743b4c5e347c866b4837f5a72  I had this problem.

I want to remind you that currently I have the active server without this changes. https://github.com/cmangos/mangos-tbc/commit/2cd35410968cc045bdefdfba7cd53c7b238f0080

It was the only way I could get rid of this problem.

**Comment by insunaa** [comment]
I also have a server running without any issues for over a week now using the patch exactly as I wrote it here.
https://github.com/cmangos/mangos-tbc/pull/441/files

Cyberium reworked the patch to make it more readable and some stuff, I haven't checked it over. On my short-lived test servers this has also always worked flawlessly.

**Comment by diluh** [comment]
@insunaa As I said I still use the old version of` item_instance `
![image](https://user-images.githubusercontent.com/46387591/112690936-57bb6880-8e5b-11eb-8c44-0b1d3ac48c67.png)

I did the following process, I used this change https://github.com/Lymphome/mangos-tbc/commit/862dafa04d5e25ec7eee20e0f92a4316c29f00f1  para converter as tabelas de` item_instance` para nova versão.

Then I applied https://github.com/cmangos/mangos-tbc/commit/2cd35410968cc045bdefdfba7cd53c7b238f0080
soon after I applied the patch of Cyberium https://github.com/cmangos/mangos-tbc/commit/5c721e3a380cb03743b4c5e347c866b4837f5a72

apparently everything was going well, more players started reporting problems with missing items.

**Comment by insunaa** [comment]
Why did you not use the included https://github.com/cmangos/mangos-tbc/tree/master/contrib/iteminstance_converter ?

**Comment by MantisLord** [comment]
The golang script should not be used anymore - it's deprecated now in favor of a regular character DB SQL update. https://github.com/cmangos/mangos-tbc/blob/master/sql/updates/characters/s2420_01_characters_item_instance_data_drop.sql

**Comment by TheOzzOne** [comment]
> 
> 
> I am also experiencing this bug; happy to help troubleshoot in any way that I can.

I've tested the fix and it appears to have solved my issues on Classic thusfar (testing 1wk / 8 characters created all with starting items).

**Comment by diluh** [comment]
@cyberium @insunaa @MantisLord 

I finally think the problem has been solved. i did the following i used this change to convert https://github.com/cmangos/mangos-tbc/blob/master/sql/updates/characters/s2420_01_characters_item_instance_data_drop.sql

1. - Clear the entire database `Auction ` tbccharacters
2. - Clear `mail `and `mail_items` -tbccharacters
3. - Changes the tables 'charges' and 'enchantments' to utf8 as shown in the image - when I did the conversion changed so I had to switch back to utf8

![image](https://user-images.githubusercontent.com/46387591/113521088-b2487900-956d-11eb-9f95-ed80f9ed2864.png)


Server is 2 days running without any error.

I think now this problem can be closed.
